crates/db_views/notification/src/impls.rs

```diff
@@ -72,14 +71,17 @@ impl NotificationView {
         .or(modlog::mod_id.eq(item_creator)),
     );
 
+    // No need to join on `modlog::target_person_id` as it is identical to
+    // `notification::recipient_id`.
     let recipient_join = aliases::person1.on(notification::recipient_id.eq(recipient_person));
 
     let comment_join = comment::table.on(
       notification::comment_id
         .eq(comment::id.nullable())
         // Filter out the deleted / removed
         .and(not(comment::deleted))
-        .and(not(comment::removed)),
+        .and(not(comment::removed))
+        .or(modlog::target_comment_id.eq(comment::id.nullable())),
```

Comment:
This line causes notifications to be duplicated in the test case. It seems to join once to `notification::comment_id` (which is null) and also to `modlog::target_comment_id` (which is not null). Whats the right way to write this so that it only joins to non-null columns?

Reply:
You can't end with an or, because it'll ignore all the previous filters.

Try moving this inside L80, IE:

`comment::id.eq(notification::comment_id.or(modlog::target_comment_id))`

Reply:
The problem is that this will exclude removed comments for modlog items.

Reply:
Then maybe do `.and(not(comment::removed).and(modlog::target_comment_id.is_null())`